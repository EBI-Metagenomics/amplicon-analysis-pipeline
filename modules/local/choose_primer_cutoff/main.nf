
process CHOOSE_PRIMER_CUTOFF {
    // Select inflection points most likely to be primer cutoff points
    tag "$meta.id"
    label 'very_light'
    container "oras://community.wave.seqera.io/library/mi-pimento:0.0.4--1795182b3e13ee89"

    input:
    tuple val(meta), path(cutoffs), path(reads_merged)

    output:
    tuple val(meta), path("*auto_primers.fasta"), emit: auto_primer_out
    path "versions.yml"                         , emit: versions

    script:
    """
    if [[ -s ./$cutoffs ]]; then
        pimento choose_primer_cutoff -i ${reads_merged} -p ${cutoffs} -o ${meta.id}_${meta.var_region}
    else
        touch ${meta.id}_${meta.var_region}_auto_primers.fasta
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mi-pimento: \$( pimento --version | cut -d" " -f3 )
    END_VERSIONS
    """

}
