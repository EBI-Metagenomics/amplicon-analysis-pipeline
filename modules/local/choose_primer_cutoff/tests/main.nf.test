nextflow_process {

    name "Test Process ASSESS_MCP_INF_POINTS"
    script "../main.nf"
    process "ASSESS_MCP_INF_POINTS"
    tag "modules"
    tag "assess_mcp_inf_points"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    file('${projectDir}/modules/local/assess_mcp_inf_points/tests/data/test_inf_points.tsv', checkIfExists: true),
                    file('${projectDir}/modules/local/assess_mcp_inf_points/tests/data/test.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.auto_primer_out).match("auto_primer_out") }
            )
        }

    }

}
