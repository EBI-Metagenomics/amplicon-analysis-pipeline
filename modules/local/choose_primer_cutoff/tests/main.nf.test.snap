{
    "auto_primer_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false,
                        "var_region": "16S-V3-V4"
                    },
                    "test_16S-V3-V4_auto_primers.fasta:md5,c36ff305d6aec14df3b7a61cb702101e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-25T11:34:07.968923891"
    }
}